core/test/utils/assertions.hpp

```diff
@@ -127,15 +134,18 @@ double get_relative_error(const MatrixData1 &first, const MatrixData2 &second)
                 get_next_value(first_it, end(first.nonzeros), row, col);
             const auto second_val =
                 get_next_value(second_it, end(second.nonzeros), row, col);
-            diff += squared_norm(first_val - second_val);
-            first_norm += squared_norm(first_val);
-            second_norm += squared_norm(second_val);
+            const auto real_first = static_cast<real_vt>(std::real(first_val));
+            const auto real_second =
+                static_cast<real_vt>(std::real(second_val));
+            diff += squared_norm(real_first - real_second);
```

Comment:
Here you are discarding the imaginary part of first and second. Is this intended behavior?

Reply:
real_first and real_second are only the real parts of first and second, so squared_norm computes only the absolute difference between the real parts, as far as I see.

Reply:
My bad, did not see the `real` call.